### ISSUE TITLE
Infinite loop during a single step

### ISSUE THREAD

**Issue by edmcman** [angr/angr#1939]
---

**Describe the bug.**

When stepping a particular program state, angr appears to get stuck in an infinite loop.

**Environment Information.**

```
angr environment report
=============================
Date: 2020-01-27 16:35:57.368022
Running in virtual environment at /home/ed/.virtualenvs/angr
Platform: linux-x86_64
Python version: 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0]
######## angr #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/angr
Pip version angr 8.20.1.7
Couldn't find git info
######## ailment #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/ailment
Pip version ailment 8.20.1.7
Couldn't find git info
######## cle #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/cle
Pip version cle 8.20.1.7
Couldn't find git info
######## pyvex #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/pyvex
Pip version pyvex 8.20.1.7
Couldn't find git info
######## claripy #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/claripy
Pip version claripy 8.20.1.7
Couldn't find git info
######## archinfo #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/archinfo
Pip version archinfo 8.20.1.7
Couldn't find git info
######## z3 #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/z3
Pip version z3-solver 4.8.7.0
Couldn't find git info
######## unicorn #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/unicorn
Pip version unicorn 1.0.1
Couldn't find git info
######### Native Module Info ##########
angr: <CDLL '/home/ed/.virtualenvs/angr/lib/python3.6/site-packages/angr/lib/angr_native.so', handle 1d6c560 at 0x7f614b1bae10>
unicorn: <CDLL '/home/ed/.virtualenvs/angr/lib/python3.6/site-packages/unicorn/lib/libunicorn.so', handle 14efe70 at 0x7f614a1c7c18>
pyvex: <cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x7f614b116dd8>
z3: <CDLL '/home/ed/.virtualenvs/angr/lib/python3.6/site-packages/z3/lib/libz3.so', handle 170cef0 at 0x7f6148011860>
```

**To Reproduce.**

``` python
import logging
logging.getLogger('angr').setLevel('DEBUG')
import angr

proj = angr.Project('/bin/true', auto_load_libs=False)
state = proj.factory.blank_state(addr=0x4011e3)
state.step()
```

**Additional context.**

This is the problematic block:
```
0x4011e3:	add	byte ptr [rip], bl
0x4011e9:	add	byte ptr [rax], al
0x4011eb:	add	byte ptr [rax], al
0x4011ed:	add	byte ptr [rax], al
0x4011ef:	add	byte ptr [rax], dh
```

angr repeats this message indefinitely:
```
DEBUG   | 2020-01-27 16:39:13,959 | angr.engines.vex.heavy.heavy | IMark: 0x4011e3
DEBUG   | 2020-01-27 16:39:13,959 | angr.state_plugins.symbolic_memory | Doing a store...
DEBUG   | 2020-01-27 16:39:13,960 | angr.state_plugins.symbolic_memory | ... done
DEBUG   | 2020-01-27 16:39:13,960 | angr.state_plugins.symbolic_memory | Doing a store...
DEBUG   | 2020-01-27 16:39:13,960 | angr.state_plugins.symbolic_memory | ... done
```

**Comment by edmcman** [comment]
Here is the problematic binary.

[true.zip](https://github.com/angr/angr/files/4119481/true.zip)

**Comment by rhelmot** [comment]
Interestingly enough, I think this was fixed by 36fe22eeea18265fec60e85bfe057d11a19a051b! The script works fine on the master branch.

**Comment by edmcman** [comment]
Awesome, thanks!
